import-scripts/study_directory_uses_updated_filenames.py

```diff
@@ -0,0 +1,538 @@
+        else:
+            filename2datatype[staging_filename] = datatype
+        if "*" in meta_filename:
+            wildcard_pattern2datatype[meta_filename] = datatype
+        else:
+            filename2datatype[meta_filename] = datatype
+    return filename2datatype, wildcard_pattern2datatype
+
+def is_normal_sample_filename(filename):
+    """
+    examines filename to see if the importer would skip it as a "normal sample" file.
+    """
+    normal_sample_suffixes = ["_normals.txt", "data_mrna_seq_v2_rsem_normal_samples.txt", "data_mrna_seq_v2_rsem_normal_samples_zscores_ref_normal_samples.txt"]
+    for suffix in normal_sample_suffixes:
+        if (filename.endswith(suffix)):
+            return True;
+    return False;
+
+def determine_cancer_study_identifier_from_meta_file(filepath):
+    """
+    search file at <filepath> for a cancer_study_identifier and return it if found
+    """
+    CANCER_STUDY_IDENTIFIER_PREFIX = "cancer_study_identifier:"
+    file = open(filepath, 'r')
+    for line in file.readlines():
+        line_without_whitespace = "".join(line.split()).strip()
+        if line.startswith(CANCER_STUDY_IDENTIFIER_PREFIX):
+            cancer_study_identifier = line[len(CANCER_STUDY_IDENTIFIER_PREFIX):].strip()
+            if len(cancer_study_identifier) > 0:
+                return cancer_study_identifier 
+    return None
+
+def determine_cancer_study_identifier(basedir, filenames):
+    """
+    search <filenames> in <basedir> for a cancer_study_identifier in any meta filename and return it if found
+    """
+    cancer_study_identifier = None
+    for filename in filenames:
+        if filename.startswith("meta_"):
+            filepath = os.path.join(basedir, filename)
+            if os.path.isfile(filepath):
+                cancer_study_identifier = determine_cancer_study_identifier_from_meta_file(filepath)
+        if cancer_study_identifier is not None:
+            # first success is adequate
+            return cancer_study_identifier
+    return None
+
+def split_wildcard_patterns(wildcard_patterns):
+    """
+    breaks a list of wildcard_patterns (each containing a single asterisk) into two parts, returning a list of dictionaries with prefix/suffix.
+    """
+    split_wildcard_pattern_list = []
+    for wildcard_pattern in wildcard_patterns:
+        pattern_segments = wildcard_pattern.strip().split("*")
+        if len(pattern_segments) != 2:
+            sys.stderr.write("Error : wildcard filename pattern '" + wildcard_pattern + "' does not have exactly one asterisk\n")
+            sys.exit(1)
+        split_wildcard_pattern = {"prefix" : pattern_segments[0], "suffix" : pattern_segments[1]}
+        split_wildcard_pattern_list.append(split_wildcard_pattern)
+    return split_wildcard_pattern_list
+
+def matches_wildcard_pattern(filename, split_wildcard_pattern):
+    """
+    returns True if filename starts with the prefix from, and ends with the suffix from, the split_wildcard_pattern dictionary.
```

Comment:
can we get some examples?

Reply:
the example added for parse_current_datatypes now is added here to show the use of the split wildcard pattern.